_posts/2018-06-15-cocoapods-keys-react-native.markdown

```diff
@@ -0,0 +1,156 @@
+
+ - (instancetype)initWithUserID:(NSString *)userID
+           authenticationToken:(NSString *)token
+                     sentryDSN:(nullable NSString *)sentryDSN
++         stripePublishableKey:(NSString *)stripePublishableKey
+              googleMapsAPIKey:(nullable NSString *)googleAPIKey
+                    gravityURL:(NSString *)gravity
+                metaphysicsURL:(NSString *)metaphysics
+                     userAgent:(NSString *)userAgent;
+```
+
+`AREmission`'s implementation (.m) needs to be configured to take this new key- It will be exported to our our React Native components as `Emission`. We make our intializer match the signature defined in the header (.h) file, and add an instance `_stripePublishableKey` to match the `@property` declaration.
+[/Pod/Classes/Core/AREmission.m](https://github.com/artsy/emission/blob/4a2a3e9260e97d791536cf38376a06b0ad0946a8/Pod/Classes/Core/AREmission.m#L24-L60):
+
+```diff
+ @implementation AREmissionConfiguration
+ RCT_EXPORT_MODULE(Emission);
+ # ...
+
+ - (NSDictionary *)constantsToExport
+ {
+   return @{
++    @"stripePublishableKey": self.stripePublishableKey ?: @"",
+     # ...lots more
+   };
+ }
+
+ - (instancetype)initWithUserID:(NSString *)userID
+            authenticationToken:(NSString *)token
+                      sentryDSN:(NSString *)sentryDSN
++          stripePublishableKey:(NSString *)stripePublishableKey
+               googleMapsAPIKey:(NSString *)googleAPIKey
+                     gravityURL:(NSString *)gravity
+                 metaphysicsURL:(NSString *)metaphysics
+                      userAgent:(nonnull NSString *)userAgent
+ {
+     self = [super init];
+     _userID = userID.copy;
++    _stripePublishableKey = stripePublishableKey.copy;
+     # ... More copies...
+     return self;
+ }
+```
+
+---
+
+#### 3. Configure the example app to expose the new key to our library
+
+Make sure we have imported they keys from `cocoapods-keys` and initialize the new configuration with the function we defined above.
+
+[Example/Emission/AppDelegate.m](https://github.com/artsy/emission/blob/4a2a3e9260e97d791536cf38376a06b0ad0946a8/Example/Emission/AppDelegate.m#L109)
+
+```objc
+#import <Keys/EmissionKeys.h>
+
+# Add the key to our initWith...  call, following the signature we defined in the previous step
+- (void)setupEmissionWithUserID:(NSString *)userID accessToken:(NSString *)accessToken keychainService:(NSString *)service;
+{
+  # ...
+
+  # stripePublishableKey is somewhere down there...
+
+  AREmissionConfiguration *config = [[AREmissionConfiguration alloc] initWithUserID:userID authenticationToken:accessToken sentryDSN:nil stripePublishableKey:[keys stripePublishableKey] googleMapsAPIKey:nil gravityURL:setup.gravityURL metaphysicsURL:setup.metaphysicsURL userAgent:@"Emission Example"];
+  # ...
+```
+
+---
+
+#### 4. Use that configured key in a `react-native` component.
+
+`Emission` is now exposed along with its configured keys via `react-native`'s `NativeModules`.
```

Comment:
might be worth a link to this - https://facebook.github.io/react-native/docs/native-modules-ios.html